libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -869,12 +873,63 @@ AP_InertialSensor::init(uint16_t loop_rate)
     // configured value
     _calculated_harmonic_notch_freq_hz[0] = _harmonic_notch_filter.center_freq_hz();
     _num_calculated_harmonic_notch_frequencies = 1;
+    _num_dynamic_harmonic_notches = 1;
 
+    uint8_t num_filters = 0;
+    const bool double_notch = _harmonic_notch_filter.hasOption(HarmonicNotchFilterParams::Options::DoubleNotch);
+#if APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane)
+    if (_harmonic_notch_filter.hasOption(HarmonicNotchFilterParams::Options::DynamicHarmonic)) {
+#if HAL_WITH_DSP
+        if (get_gyro_harmonic_notch_tracking_mode() == HarmonicNotchDynamicMode::UpdateGyroFFT) {
+            _num_dynamic_harmonic_notches = AP_HAL::DSP::MAX_TRACKED_PEAKS; // only 3 peaks supported currently
+        } else
+#endif
+        {
+            AP_Motors *motors = AP::motors();
+            _num_dynamic_harmonic_notches = __builtin_popcount(motors->get_motor_mask());
+        }
+        // avoid harmonics unless actually configured by the user
+        _harmonic_notch_filter.set_default_harmonics(1);
+    }
+#endif
+    // calculate number of notches we might want to use for harmonic notch
+    if (_harmonic_notch_filter.enabled()) {
+        num_filters = __builtin_popcount( _harmonic_notch_filter.harmonics())
+            * _num_dynamic_harmonic_notches * (double_notch ? 2 : 1)
+            * (sensor_used(0) + sensor_used(1) + sensor_used(2)
+            * (1 << _fast_sampling_rate)); // each increment in gyro rate halves the number of notches that can be used
+    }
+    // add filters used by static notch
+    if (_notch_filter.enabled()) {
+        _notch_filter.set_default_harmonics(1);
+        num_filters +=  __builtin_popcount(_notch_filter.harmonics())
+            * (_notch_filter.hasOption(HarmonicNotchFilterParams::Options::DoubleNotch) ? 2 : 1)
+            * (sensor_used(0) + sensor_used(1) + sensor_used(2)
+            * (1 << _fast_sampling_rate)); // each increment in gyro rate halves the number of notches that can be used
+    }
+
+    if (num_filters > (HAL_HNF_MAX_FILTERS * INS_MAX_INSTANCES)) {
+        AP_BoardConfig::config_error("Too many notches: %u > %u", num_filters, (HAL_HNF_MAX_FILTERS * INS_MAX_INSTANCES));
```

Comment:
try triggering this, and ensure that a MissionPlanner user can see whats happening

Reply:
This shows up in Mission Planner